semcore/portal/src/Portal.tsx

```diff
@@ -21,20 +23,40 @@ const PortalContext = register.get(
   React.createContext<NodeByRef>((canUseDOM() ? document.body : null) as any),
 );
 
+const key = 'intergalactic-portal-prepend-mount-node';
+const prependMountNode = (mountNode: Element) => {
+  const firstChildren = mountNode.children[0];
+  if ((firstChildren as HTMLElement)?.dataset?.portal === key) return firstChildren as HTMLElement;
+  const node = document.createElement('div');
+  node.dataset.portal = key;
+  mountNode.prepend(node);
+  return node as HTMLElement;
+};
+
 function Portal(props: PortalProps & { Children: React.FC }) {
-  const { Children, disablePortal, ignorePortalsStacking, onMount } = props;
+  const { Children, disablePortal, ignorePortalsStacking, onMount, portalRendering } = props;
   const container = React.useContext(PortalContext);
   const [mountNode, setMountNode] = React.useState<Element | null>(null);
 
   React.useEffect(() => {
     if (disablePortal) return;
     onMount?.(true);
     if (ignorePortalsStacking) {
-      setMountNode(canUseDOM() ? document.body : null);
+      if (!canUseDOM()) return;
+      let mountNode = document.body;
+      if (portalRendering === 'prepend') {
+        mountNode = prependMountNode(mountNode);
+      }
+      setMountNode(document.body);
```

Comment:
`setMountNode(mountNode)` maybe ?

And mountNode in function Portal and mountNode in effect adds ambiguity...